### PR TITLE
Added Bounds for Lists.

### PR DESCRIPTION
**Pull request by jfdm**
A direct copy from the Idris1 version.

### CODE REVIEW

**Comment by melted** [comment]
Just the expected output from a test that needs to be updated

**Comment by jfdm** [comment]
How on earth did my PR result in a test failure!

**Comment by melted** [comment]
Reflection003 is leaking some internal number into the output, you just have to update that expected file.

**Comment by jfdm** [comment]
That is interesting, locally the test is passing...

**Comment by gallais** [comment]
> Reflection003 is leaking some internal number into the output, you just have to update that expected file.

Sounds like we should have a little `sed` cleanup phase.

**Comment by edwinb** [comment]
There's a couple of tests that leak internal details, or sometimes line numbers. In some cases, we should see if we can tidy up the output (like case and with blocks). The reflection one could probably be rewritten just to check that the generated name isn't "plus".

**Comment by jfdm** [comment]
urgh, that is annoying. I am not sure how best to work around this issue given that the tests do not fail locally....

**Comment by jfdm** [comment]
I've rebased on master so let us see what happens...

File: libs/base/Data/List.idr [hunk 1]
```diff
@@ -359,6 +359,40 @@ init' : List a -> Maybe (List a)
...
+public export
+data InBounds : (k : Nat) -> (xs : List a) -> Type where
+    ||| Z is a valid index into any cons cell
+    InFirst : InBounds Z (x :: xs)
+    ||| Valid indices can be extended
+    InLater : InBounds k xs -> InBounds (S k) (x :: xs)
+
+export
+Uninhabited (InBounds k []) where
+    uninhabited InFirst impossible
```
[the remark this answers is not in the document]
Comment (jfdm) [reply libs/base/Data/List.idr:375]: that is a relic from the Idris1 definition IIRC, I have fixed it locally. Neither of the compilers complained about it ;-)

**Comment by jfdm** [comment]
Right, this is annoying me too much. I am going to abandon ship and start this PR anew....

**Pull request closed without merging**